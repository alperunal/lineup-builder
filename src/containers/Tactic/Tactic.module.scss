@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300&display=swap');

.tactic {
    padding: 25px 0 125px 0;

    &__tactic-field {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
        margin-top: 50px;
        flex-direction: column;

        @media(min-width: 1000px) {
            flex-direction: row;
        }
    }

    &__field-wrapper {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        position: relative;

        @media(min-width: 1000px) {
            width: 50%;
        }
    }

    &__actions {
        display: flex;
        flex-wrap: wrap;
        margin: 0;
        list-style: none;
        padding: 0;
        position: absolute;
        right: 0;
        margin-right: 80px;
        top: 4px;

        li {
            border: 0;
            margin-left: 4px;
        }

        button {
            background: linear-gradient(135deg, #05ac50 0%,#21dd72 50%,#05c44e 51%,#05ac50 71%,#05ac50 100%);;
            padding: 10px;
            border: 0;
            cursor: pointer;
            border-radius: 50%;
            width: 36px;
            opacity: .5;
            box-shadow: 3px 3px 3px;

            &:hover {
                opacity: 1;
            }
        }
    }

    .heading {
        font-size: 3em;
        color: #181c32;
    }

    .options {
        width: 100%;
        box-shadow: 0 0 30px 0 rgba(82,63,105,.05);
        border: 0;
        border-radius: .42rem;
        font-family: 'Poppins', sans-serif;

        @media(min-width: 1000px) {
            width: 50%;
        }

        &__header {
            min-height: 70px;
            padding: 0 2.25rem;
            margin-bottom: 0;
            border-bottom: 1px solid #ebedf3;
            display: flex;
            align-items: center;
        }

        &__content {
            padding: 2.25rem;
        }
    }
    
    .color-options {
        display: flex;
        justify-content: space-around;
    }
    
    .color-option {
        flex: 1;
        display: flex;
        flex-direction: column;
        align-items: center;
    }
}