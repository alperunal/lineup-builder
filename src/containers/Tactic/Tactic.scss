.tactic {
    &__tactic-field {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
    }

    &__field-wrapper {
        width: 400px;
    }

    &__options {
        background-color: rgba(0, 0, 0, 0.5);
        border-radius: 0.5rem;
        display: block;
        box-shadow: 0 4px 25px 0 rgba(0, 0, 0, 0.1);
        transition: all 0.3s ease-in-out;
        margin-bottom: 2rem;
        margin-top: 20px;

        @media (min-width: 768px) {
            margin-top: 25px;
        }
    }

    &__color-options {
        display: flex;
        justify-content: space-around;
    }

    &__color-option {
        flex: 1;
        display: flex;
        flex-direction: column;
        align-items: center;
    }

    &__label {
        padding-left: 5px;
        color: #fff;
        margin-bottom: 5px;
        text-shadow: 2px 2px #000;
    }

    &__input {
        padding-left: 10px;
        height: 40px;
        background: rgba(0, 0, 0, 0.5);
        color: #fff;
        border: 4px solid rgba(0, 0, 0, 0.1);
        border-radius: 5px;
        font-family: 'Roboto';
        font-size: 20px;
        box-sizing: border-box;
        margin-bottom: 5px;

        &::placeholder {
            color: #ddd;
        }
    }

    &__form-field {
        margin-top: 1.5rem;
        position: relative;
        transition: all 0.3s ease;
        display: flex;
        flex-direction: column;
        padding: 10px;
    }
}