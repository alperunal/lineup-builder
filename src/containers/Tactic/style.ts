import styled from "styled-components";

export const TacticField = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

export const Options = styled.div`
  background-color: rgba(0, 0, 0, 0.5);
  border-radius: 0.5rem;
  display: block;
  box-shadow: 0 4px 25px 0 rgba(0, 0, 0, 0.1);
  transition: all 0.3s ease-in-out;
  margin-bottom: 2rem;
  margin-top: 20px;

  @media(min-width: 768px) {
    margin-top: 0;
  }
`;

export const ColorOptions = styled.div`
  display: flex;
  justify-content: space-around;
`;

export const ColorOption = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Label = styled.label`
  padding-left: 5px;
  color: #fff;
  margin-bottom: 5px;
  text-shadow: 2px 2px #000;
`;

export const Input = styled.input`
  padding-left: 10px;
  height: 40px;
  background: rgba(0, 0, 0, 0.5);
  color: #fff;
  border: 4px solid rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  font-family: "Roboto";
  font-size: 20px;
  box-sizing: border-box;
  margin-bottom: 5px;

  &::placeholder {
    color: #ddd;
  }
`;

export const FormField = styled.div`
  margin-top: 1.5rem;
  position: relative;
  transition: all 0.3s ease;
  display: flex;
  flex-direction: column;
  padding: 10px;
`;
