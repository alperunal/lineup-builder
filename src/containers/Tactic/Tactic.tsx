import React, { useState } from 'react';

import ColorPicker from '../../components/UI/ColorPicker/ColorPicker';
import Field from '../../components/Field/Field';
import Squad from '../../components/Squad/Squad';
import { IPosition } from '../../constants/model';
import { GROUND } from '../../constants/constants';
import {generateData} from '../../helpers/player-generator';
import './Tactic.scss';
import { Card, Input } from '../../style/VoetUI';

const Tactic = () => {
    const [name, setName] = useState('Default');
    const [mainColor, setMainColor] = useState('#a32638');
    const [secondaryColor, setSecondaryColor] = useState('#fcb514');
    const [numberColor, setNumberColor] = useState('#ffffff');
    const [players, setPlayers] = useState(generateData());

    function editPlayer(id: string, nName: string, nNum: number, nPosition: IPosition) {
        setPlayers(
            players.map(player => {
                if (player.id === id) {
                    return {
                        name: nName !== null ? nName : player.name,
                        num: nNum !== null ? nNum : player.num,
                        id,
                        position: nPosition !== null ? nPosition : player.position,
                    };
                } else {
                    return player;
                }
            }),
        );
    }

    function handleDrop(event: any) {
        // drop event handler
    }
    
    return (
        <div className="tactic">
            <div className="tactic__tactic-field">
                <Card customClass="tactic__field-wrapper">
                    <Field
                        width={GROUND.width}
                        height={GROUND.height}
                        players={players}
                        mainColor={mainColor}
                        secondaryColor={secondaryColor}
                        numberColor={numberColor}
                        handleDrop={handleDrop}
                    />
                </Card>
                <div className="tactic__options">
                    <div className="tactic__form-field">
                        <label className="tactic__label" htmlFor="tacticName">Tactic Name</label>
                        <Input className="tactic__input" id="tacticName" type="text" value={name} onChange={event => setName(event.target.value)} />
                    </div>
                    <div className="tactic__color-options">
                        <div className="tactic__color-option">
                            <label className="tactic__label">Primary</label>
                            <ColorPicker color={mainColor} setColor={setMainColor} />
                        </div>
                        <div className="tactic__color-option">
                            <label className="tactic__label">Secondary</label>
                            <ColorPicker color={secondaryColor} setColor={setSecondaryColor} />
                        </div>
                        <div className="tactic__color-option">
                            <label className="tactic__label">Number</label>
                            <ColorPicker color={numberColor} setColor={setNumberColor} />
                        </div>
                    </div>
                    <Squad players={players} editPlayer={editPlayer} />
                </div>
            </div>
        </div>
    );
};

export default Tactic;
