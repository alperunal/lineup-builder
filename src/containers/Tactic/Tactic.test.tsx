import React from 'react';
import { shallow } from 'enzyme';
import Tactic from './Tactic';

it('renders Container correctly', () => {
    const wrapper = shallow(<Tactic />);

    expect(wrapper).toMatchSnapshot();
});
