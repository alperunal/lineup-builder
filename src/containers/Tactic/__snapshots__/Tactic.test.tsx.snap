// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Tactic correctly 1`] = `ShallowWrapper {}`;
